src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx

```diff
@@ -667,17 +658,26 @@ function reorderData(
 ): BarplotDataWithStatistics | BarplotData {
   // If faceted, reorder the facets and within the facets
   if (isFaceted(data)) {
+    // for each value in the facet vocabulary's correct order
+    // find the index in the series where series.name equals that value
+    const facetValues = data.facets.map((facet) => facet.label);
+    const facetIndices = facetVocabulary.map((name) =>
+      facetValues.indexOf(name)
+    );
+
     return {
       ...data,
-      facets: sortBy(data.facets, ({ label }) =>
-        facetVocabulary.indexOf(label)
-      ).map(({ label, data }) => ({
-        label,
-        data: reorderData(
-          data,
-          labelVocabulary,
-          overlayVocabulary
-        ) as BarplotData,
+      facets: facetIndices.map((i, j) => ({
+        label: facetVocabulary[j],
+        data:
+          data.facets[i]?.data != null
+            ? (reorderData(
+                data.facets[i].data!, // not sure why ! is needed
```

Comment:
I'd appreciate any wisdom on this!

Reply:
I don't think typescript can guarantee that `i` hasn't been mutated since the not-null assertion on the line above, so it can't assume the expression still evaluates to not-null.

You can avoid the `!` assertion by storing `data.facets[i].data` in a variable.

Reply:
Thanks! That works great, and makes it more readable.